
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Natural Products Linker">
      
      
      
        <link rel="canonical" href="https://nplinker.github.io/nplinker/dev/api/genomics/">
      
      
        <link rel="prev" href="../loader/">
      
      
        <link rel="next" href="../genomics_abc/">
      
      
      <link rel="icon" href="../../favicon.png">
      <meta name="generator" content="mkdocs-1.5.3, mkdocs-material-9.5.12">
    
    
      
        <title>Data Models - NPLinker</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.7e359304.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../assets/_mkdocstrings.css">
    
      <link rel="stylesheet" href="../../extra/terminal.css">
    
      <link rel="stylesheet" href="../../extra/tweaks.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="deep-orange" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#nplinker.genomics" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
      </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="NPLinker" class="md-header__button md-logo" aria-label="NPLinker" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 14.27 10.64 13A11.242 11.242 0 0 0 5 10.18v6.95c2.61.34 5 1.34 7 2.82 2-1.48 4.39-2.48 7-2.82v-6.95c-2.16.39-4.09 1.39-5.64 2.82M19 8.15c.65-.1 1.32-.15 2-.15v11c-3.5 0-6.64 1.35-9 3.54C9.64 20.35 6.5 19 3 19V8c.68 0 1.35.05 2 .15 2.69.41 5.1 1.63 7 3.39 1.9-1.76 4.31-2.98 7-3.39M12 6c.27 0 .5-.1.71-.29.19-.21.29-.44.29-.71s-.1-.5-.29-.71C12.5 4.11 12.27 4 12 4s-.5.11-.71.29c-.18.21-.29.45-.29.71s.11.5.29.71c.21.19.45.29.71.29m2.12 1.12a2.997 2.997 0 1 1-4.24-4.24 2.997 2.997 0 1 1 4.24 4.24Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            NPLinker
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Data Models
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="deep-orange" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3m0-7 2.39 3.42C13.65 5.15 12.84 5 12 5c-.84 0-1.65.15-2.39.42L12 2M3.34 7l4.16-.35A7.2 7.2 0 0 0 5.94 8.5c-.44.74-.69 1.5-.83 2.29L3.34 7m.02 10 1.76-3.77a7.131 7.131 0 0 0 2.38 4.14L3.36 17M20.65 7l-1.77 3.79a7.023 7.023 0 0 0-2.38-4.15l4.15.36m-.01 10-4.14.36c.59-.51 1.12-1.14 1.54-1.86.42-.73.69-1.5.83-2.29L20.64 17M12 22l-2.41-3.44c.74.27 1.55.44 2.41.44.82 0 1.63-.17 2.37-.44L12 22Z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="deep-orange" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m17.75 4.09-2.53 1.94.91 3.06-2.63-1.81-2.63 1.81.91-3.06-2.53-1.94L12.44 4l1.06-3 1.06 3 3.19.09m3.5 6.91-1.64 1.25.59 1.98-1.7-1.17-1.7 1.17.59-1.98L15.75 11l2.06-.05L18.5 9l.69 1.95 2.06.05m-2.28 4.95c.83-.08 1.72 1.1 1.19 1.85-.32.45-.66.87-1.08 1.27C15.17 23 8.84 23 4.94 19.07c-3.91-3.9-3.91-10.24 0-14.14.4-.4.82-.76 1.27-1.08.75-.53 1.93.36 1.85 1.19-.27 2.86.69 5.83 2.89 8.02a9.96 9.96 0 0 0 8.02 2.89m-1.64 2.02a12.08 12.08 0 0 1-7.8-3.47c-2.17-2.19-3.33-5-3.49-7.82-2.81 3.14-2.7 7.96.31 10.98 3.02 3.01 7.84 3.12 10.98.31Z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var media,input,key,value,palette=__md_get("__palette");if(palette&&palette.color){"(prefers-color-scheme)"===palette.color.media&&(media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']"),palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent"));for([key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512"><!--! Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M41.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 256l137.3-137.4c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/nplinker/nplinker" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    nplinker/nplinker
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../.." class="md-tabs__link">
          
  
  Get Started

        </a>
      </li>
    
  

      
        
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../nplinker/" class="md-tabs__link">
          
  
  API Documentation

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="NPLinker" class="md-nav__button md-logo" aria-label="NPLinker" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 14.27 10.64 13A11.242 11.242 0 0 0 5 10.18v6.95c2.61.34 5 1.34 7 2.82 2-1.48 4.39-2.48 7-2.82v-6.95c-2.16.39-4.09 1.39-5.64 2.82M19 8.15c.65-.1 1.32-.15 2-.15v11c-3.5 0-6.64 1.35-9 3.54C9.64 20.35 6.5 19 3 19V8c.68 0 1.35.05 2 .15 2.69.41 5.1 1.63 7 3.39 1.9-1.76 4.31-2.98 7-3.39M12 6c.27 0 .5-.1.71-.29.19-.21.29-.44.29-.71s-.1-.5-.29-.71C12.5 4.11 12.27 4 12 4s-.5.11-.71.29c-.18.21-.29.45-.29.71s.11.5.29.71c.21.19.45.29.71.29m2.12 1.12a2.997 2.997 0 1 1-4.24-4.24 2.997 2.997 0 1 1 4.24 4.24Z"/></svg>

    </a>
    NPLinker
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/nplinker/nplinker" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    nplinker/nplinker
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" >
        
          
          <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Get Started
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1">
            <span class="md-nav__icon md-icon"></span>
            Get Started
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Welcome to NPLinker
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
      
        
        
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" checked>
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="">
            
  
  <span class="md-ellipsis">
    API Documentation
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            API Documentation
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../nplinker/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    NPLinker
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../arranger/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Dataset Arranger
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../loader/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Dataset Loader
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_4" checked>
        
          
          <label class="md-nav__link" for="__nav_2_4" id="__nav_2_4_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Genomics Data
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_4_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2_4">
            <span class="md-nav__icon md-icon"></span>
            Genomics Data
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Data Models
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Data Models
  </span>
  

      </a>
      
        

  

<nav class="md-nav md-nav--secondary" aria-label="Page contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Page contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#nplinker.genomics" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-module"></code>&nbsp;genomics
    </span>
  </a>
  
    <nav class="md-nav" aria-label=" genomics">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.BGC" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-class"></code>&nbsp;BGC
    </span>
  </a>
  
    <nav class="md-nav" aria-label=" BGC">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.BGC.bigscape_classes" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-attribute"></code>&nbsp;bigscape_classes
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.BGC.aa_predictions" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-attribute"></code>&nbsp;aa_predictions
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.BGC.add_parent" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-method"></code>&nbsp;add_parent
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.BGC.detach_parent" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-method"></code>&nbsp;detach_parent
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.BGC.is_mibig" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-method"></code>&nbsp;is_mibig
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.GCF" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-class"></code>&nbsp;GCF
    </span>
  </a>
  
    <nav class="md-nav" aria-label=" GCF">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.GCF.bgcs" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-attribute"></code>&nbsp;bgcs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.GCF.strains" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-attribute"></code>&nbsp;strains
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.GCF.add_bgc" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-method"></code>&nbsp;add_bgc
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.GCF.detach_bgc" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-method"></code>&nbsp;detach_bgc
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.GCF.has_strain" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-method"></code>&nbsp;has_strain
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.GCF.has_mibig_only" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-method"></code>&nbsp;has_mibig_only
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.GCF.is_singleton" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-method"></code>&nbsp;is_singleton
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../genomics_abc/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Base Classes
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../mibig/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    MiBIG
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../antismash/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    AntiSMASH
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bigscape/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    BigScape
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../genomics_utils/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Utilities
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_5" >
        
          
          <label class="md-nav__link" for="__nav_2_5" id="__nav_2_5_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Metabolomics Data
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2_5">
            <span class="md-nav__icon md-icon"></span>
            Metabolomics Data
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../metabolomics/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Data Models
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../metabolomics_abc/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Base Classes
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../gnps/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    GNPS
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../metabolomics_utils/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Utilities
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_6" >
        
          
          <label class="md-nav__link" for="__nav_2_6" id="__nav_2_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Strain Data
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2_6">
            <span class="md-nav__icon md-icon"></span>
            Strain Data
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../strain/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Data Models
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../strain_utils/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Utilities
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../scoring/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Scoring
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../schema/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Schemas
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../utils/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    General Utilities
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  

<nav class="md-nav md-nav--secondary" aria-label="Page contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Page contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#nplinker.genomics" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-module"></code>&nbsp;genomics
    </span>
  </a>
  
    <nav class="md-nav" aria-label=" genomics">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.BGC" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-class"></code>&nbsp;BGC
    </span>
  </a>
  
    <nav class="md-nav" aria-label=" BGC">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.BGC.bigscape_classes" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-attribute"></code>&nbsp;bigscape_classes
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.BGC.aa_predictions" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-attribute"></code>&nbsp;aa_predictions
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.BGC.add_parent" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-method"></code>&nbsp;add_parent
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.BGC.detach_parent" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-method"></code>&nbsp;detach_parent
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.BGC.is_mibig" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-method"></code>&nbsp;is_mibig
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.GCF" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-class"></code>&nbsp;GCF
    </span>
  </a>
  
    <nav class="md-nav" aria-label=" GCF">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.GCF.bgcs" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-attribute"></code>&nbsp;bgcs
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.GCF.strains" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-attribute"></code>&nbsp;strains
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.GCF.add_bgc" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-method"></code>&nbsp;add_bgc
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.GCF.detach_bgc" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-method"></code>&nbsp;detach_bgc
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.GCF.has_strain" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-method"></code>&nbsp;has_strain
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.GCF.has_mibig_only" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-method"></code>&nbsp;has_mibig_only
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#nplinker.genomics.GCF.is_singleton" class="md-nav__link">
    <span class="md-ellipsis">
      <code class="doc-symbol doc-symbol-toc doc-symbol-method"></code>&nbsp;is_singleton
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  <h1>Data Models</h1>

<div class="doc doc-object doc-module">



<h2 id="nplinker.genomics" class="doc doc-heading">
<code class="doc-symbol doc-symbol-heading doc-symbol-module"></code>          <span class="doc doc-object-name doc-module-name">genomics</span>


<a href="#nplinker.genomics" class="headerlink" title="Permanent link">&para;</a></h2>

  <div class="doc doc-contents first">

  

  <div class="doc doc-children">








<div class="doc doc-object doc-class">



<h3 id="nplinker.genomics.BGC" class="doc doc-heading">
<code class="doc-symbol doc-symbol-heading doc-symbol-class"></code>          <span class="doc doc-object-name doc-class-name">BGC</span>


<a href="#nplinker.genomics.BGC" class="headerlink" title="Permanent link">&para;</a></h3>
<div class="language-python doc-signature highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="n">BGC</span><span class="p">(</span><span class="n">bgc_id</span><span class="p">:</span> <span class="n"><a class="autorefs autorefs-external" href="https://docs.python.org/3/library/stdtypes.html#str">str</a></span><span class="p">,</span> <span class="o">/</span><span class="p">,</span> <span class="o">*</span><span class="n">product_prediction</span><span class="p">:</span> <span class="n"><a class="autorefs autorefs-external" href="https://docs.python.org/3/library/stdtypes.html#str">str</a></span><span class="p">)</span>
</code></pre></div>

  <div class="doc doc-contents ">

  
      <p>Class to model BGC (biosynthetic gene cluster) data.</p>
<p>BGC data include both annotations and sequence data. This class is
mainly designed to model the annotations or metadata.</p>
<p>The raw BGC data is stored in GenBank format (.gbk). Additional
<code>GenBank features</code>_ could be added to the GenBank file to annotate
BGCs, e.g. antiSMASH has some self-defined features (like "region") in
its output GenBank files.</p>
<p>The annotations of BGC can be stored in JSON format, which is defined
and used by MIBiG.</p>



  <p><strong>Parameters:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Name</th>
        <th>Type</th>
        <th>Description</th>
        <th>Default</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td><code>bgc_id</code></td>
          <td>
                <code><a class="autorefs autorefs-external" href="https://docs.python.org/3/library/stdtypes.html#str">str</a></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>BGC identifier, e.g. MIBiG accession, GenBank accession.</p>
            </div>
          </td>
          <td>
              <em>required</em>
          </td>
        </tr>
        <tr>
          <td><code>product_prediction</code></td>
          <td>
                <code><a class="autorefs autorefs-external" href="https://docs.python.org/3/library/stdtypes.html#str">str</a></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>BGC's (predicted) natural products
or product classes.</p>
            </div>
          </td>
          <td>
                <code>()</code>
          </td>
        </tr>
    </tbody>
  </table>



  <p><strong>Attributes:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Name</th>
        <th>Type</th>
        <th>Description</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td><code><span title="nplinker.genomics.BGC.bgc_id">bgc_id</span></code></td>
          <td>
          </td>
          <td>
            <div class="doc-md-description">
              <p>BGC identifier, e.g. MIBiG accession, GenBank accession.</p>
            </div>
          </td>
        </tr>
        <tr>
          <td><code><span title="nplinker.genomics.BGC.product_prediction">product_prediction</span></code></td>
          <td>
          </td>
          <td>
            <div class="doc-md-description">
              <p>A tuple of (predicted) natural
products or product classes of the BGC.
For antiSMASH's GenBank data, the feature <code>region /product</code>
gives product information.
For MIBiG metadata, its biosynthetic class provides such info.</p>
            </div>
          </td>
        </tr>
        <tr>
          <td><code><span title="nplinker.genomics.BGC.mibig_bgc_class">mibig_bgc_class</span></code></td>
          <td>
          </td>
          <td>
            <div class="doc-md-description">
              <p>A tuple of MIBiG biosynthetic
classes to which the BGC belongs.
Defaults to None.
MIBiG defines 6 major biosynthetic classes for natural products,
including "NRP", "Polyketide", "RiPP", "Terpene", "Saccharide"
and "Alkaloid". Note that natural products created by all other
biosynthetic mechanisms fall under the category "Other".
More details see the publication: <a href="https://doi.org/10.1186/s40793-018-0318-y">https://doi.org/10.1186/s40793-018-0318-y</a>.</p>
            </div>
          </td>
        </tr>
        <tr>
          <td><code><span title="nplinker.genomics.BGC.description">description</span></code></td>
          <td>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Brief description of the BGC.
Defaults to None.</p>
            </div>
          </td>
        </tr>
        <tr>
          <td><code><span title="nplinker.genomics.BGC.smiles">smiles</span></code></td>
          <td>
          </td>
          <td>
            <div class="doc-md-description">
              <p>A tuple of SMILES formulas of the BGC's
products.
Defaults to None.</p>
            </div>
          </td>
        </tr>
        <tr>
          <td><code><span title="nplinker.genomics.BGC.antismash_file">antismash_file</span></code></td>
          <td>
          </td>
          <td>
            <div class="doc-md-description">
              <p>The path to the antiSMASH GenBank file.
Defaults to None.</p>
            </div>
          </td>
        </tr>
        <tr>
          <td><code><span title="nplinker.genomics.BGC.antismash_id">antismash_id</span></code></td>
          <td>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Identifier of the antiSMASH BGC, referring
to the feature <code>VERSION</code> of GenBank file.
Defaults to None.</p>
            </div>
          </td>
        </tr>
        <tr>
          <td><code><span title="nplinker.genomics.BGC.antismash_region">antismash_region</span></code></td>
          <td>
          </td>
          <td>
            <div class="doc-md-description">
              <p>AntiSMASH BGC region number, referring
to the feature <code>region</code> of GenBank file.
Defaults to None.</p>
            </div>
          </td>
        </tr>
        <tr>
          <td><code><span title="nplinker.genomics.BGC.parents">parents</span></code></td>
          <td>
          </td>
          <td>
            <div class="doc-md-description">
              <p>The set of GCFs that contain the BGC.</p>
            </div>
          </td>
        </tr>
        <tr>
          <td><code><span title="nplinker.genomics.BGC.strain">strain</span></code></td>
          <td>
          </td>
          <td>
            <div class="doc-md-description">
              <p>The strain of the BGC.</p>
            </div>
          </td>
        </tr>
    </tbody>
  </table>
      <p>.. GenBank features:
    <a href="https://www.insdc.org/submitting-standards/feature-table/">https://www.insdc.org/submitting-standards/feature-table/</a></p>

                <details class="quote">
                  <summary>Source code in <code>src/nplinker/genomics/bgc.py</code></summary>
                  <div class="language-python highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-0-16">16</a></span>
<span class="normal"><a href="#__codelineno-0-17">17</a></span>
<span class="normal"><a href="#__codelineno-0-18">18</a></span>
<span class="normal"><a href="#__codelineno-0-19">19</a></span>
<span class="normal"><a href="#__codelineno-0-20">20</a></span>
<span class="normal"><a href="#__codelineno-0-21">21</a></span>
<span class="normal"><a href="#__codelineno-0-22">22</a></span>
<span class="normal"><a href="#__codelineno-0-23">23</a></span>
<span class="normal"><a href="#__codelineno-0-24">24</a></span>
<span class="normal"><a href="#__codelineno-0-25">25</a></span>
<span class="normal"><a href="#__codelineno-0-26">26</a></span>
<span class="normal"><a href="#__codelineno-0-27">27</a></span>
<span class="normal"><a href="#__codelineno-0-28">28</a></span>
<span class="normal"><a href="#__codelineno-0-29">29</a></span>
<span class="normal"><a href="#__codelineno-0-30">30</a></span>
<span class="normal"><a href="#__codelineno-0-31">31</a></span>
<span class="normal"><a href="#__codelineno-0-32">32</a></span>
<span class="normal"><a href="#__codelineno-0-33">33</a></span>
<span class="normal"><a href="#__codelineno-0-34">34</a></span>
<span class="normal"><a href="#__codelineno-0-35">35</a></span>
<span class="normal"><a href="#__codelineno-0-36">36</a></span>
<span class="normal"><a href="#__codelineno-0-37">37</a></span>
<span class="normal"><a href="#__codelineno-0-38">38</a></span>
<span class="normal"><a href="#__codelineno-0-39">39</a></span>
<span class="normal"><a href="#__codelineno-0-40">40</a></span>
<span class="normal"><a href="#__codelineno-0-41">41</a></span>
<span class="normal"><a href="#__codelineno-0-42">42</a></span>
<span class="normal"><a href="#__codelineno-0-43">43</a></span>
<span class="normal"><a href="#__codelineno-0-44">44</a></span>
<span class="normal"><a href="#__codelineno-0-45">45</a></span>
<span class="normal"><a href="#__codelineno-0-46">46</a></span>
<span class="normal"><a href="#__codelineno-0-47">47</a></span>
<span class="normal"><a href="#__codelineno-0-48">48</a></span>
<span class="normal"><a href="#__codelineno-0-49">49</a></span>
<span class="normal"><a href="#__codelineno-0-50">50</a></span>
<span class="normal"><a href="#__codelineno-0-51">51</a></span>
<span class="normal"><a href="#__codelineno-0-52">52</a></span>
<span class="normal"><a href="#__codelineno-0-53">53</a></span>
<span class="normal"><a href="#__codelineno-0-54">54</a></span>
<span class="normal"><a href="#__codelineno-0-55">55</a></span>
<span class="normal"><a href="#__codelineno-0-56">56</a></span>
<span class="normal"><a href="#__codelineno-0-57">57</a></span>
<span class="normal"><a href="#__codelineno-0-58">58</a></span>
<span class="normal"><a href="#__codelineno-0-59">59</a></span>
<span class="normal"><a href="#__codelineno-0-60">60</a></span>
<span class="normal"><a href="#__codelineno-0-61">61</a></span>
<span class="normal"><a href="#__codelineno-0-62">62</a></span>
<span class="normal"><a href="#__codelineno-0-63">63</a></span>
<span class="normal"><a href="#__codelineno-0-64">64</a></span>
<span class="normal"><a href="#__codelineno-0-65">65</a></span>
<span class="normal"><a href="#__codelineno-0-66">66</a></span>
<span class="normal"><a href="#__codelineno-0-67">67</a></span>
<span class="normal"><a href="#__codelineno-0-68">68</a></span>
<span class="normal"><a href="#__codelineno-0-69">69</a></span>
<span class="normal"><a href="#__codelineno-0-70">70</a></span>
<span class="normal"><a href="#__codelineno-0-71">71</a></span>
<span class="normal"><a href="#__codelineno-0-72">72</a></span>
<span class="normal"><a href="#__codelineno-0-73">73</a></span>
<span class="normal"><a href="#__codelineno-0-74">74</a></span>
<span class="normal"><a href="#__codelineno-0-75">75</a></span>
<span class="normal"><a href="#__codelineno-0-76">76</a></span>
<span class="normal"><a href="#__codelineno-0-77">77</a></span>
<span class="normal"><a href="#__codelineno-0-78">78</a></span>
<span class="normal"><a href="#__codelineno-0-79">79</a></span>
<span class="normal"><a href="#__codelineno-0-80">80</a></span>
<span class="normal"><a href="#__codelineno-0-81">81</a></span>
<span class="normal"><a href="#__codelineno-0-82">82</a></span>
<span class="normal"><a href="#__codelineno-0-83">83</a></span>
<span class="normal"><a href="#__codelineno-0-84">84</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-0-16" name="__codelineno-0-16"></a><span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">bgc_id</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="o">/</span><span class="p">,</span> <span class="o">*</span><span class="n">product_prediction</span><span class="p">:</span> <span class="nb">str</span><span class="p">):</span>
<a id="__codelineno-0-17" name="__codelineno-0-17"></a><span class="w">    </span><span class="sd">&quot;&quot;&quot;Class to model BGC (biosynthetic gene cluster) data.</span>
<a id="__codelineno-0-18" name="__codelineno-0-18"></a>
<a id="__codelineno-0-19" name="__codelineno-0-19"></a><span class="sd">    BGC data include both annotations and sequence data. This class is</span>
<a id="__codelineno-0-20" name="__codelineno-0-20"></a><span class="sd">    mainly designed to model the annotations or metadata.</span>
<a id="__codelineno-0-21" name="__codelineno-0-21"></a>
<a id="__codelineno-0-22" name="__codelineno-0-22"></a><span class="sd">    The raw BGC data is stored in GenBank format (.gbk). Additional</span>
<a id="__codelineno-0-23" name="__codelineno-0-23"></a><span class="sd">    `GenBank features`_ could be added to the GenBank file to annotate</span>
<a id="__codelineno-0-24" name="__codelineno-0-24"></a><span class="sd">    BGCs, e.g. antiSMASH has some self-defined features (like &quot;region&quot;) in</span>
<a id="__codelineno-0-25" name="__codelineno-0-25"></a><span class="sd">    its output GenBank files.</span>
<a id="__codelineno-0-26" name="__codelineno-0-26"></a>
<a id="__codelineno-0-27" name="__codelineno-0-27"></a><span class="sd">    The annotations of BGC can be stored in JSON format, which is defined</span>
<a id="__codelineno-0-28" name="__codelineno-0-28"></a><span class="sd">    and used by MIBiG.</span>
<a id="__codelineno-0-29" name="__codelineno-0-29"></a>
<a id="__codelineno-0-30" name="__codelineno-0-30"></a><span class="sd">    Args:</span>
<a id="__codelineno-0-31" name="__codelineno-0-31"></a><span class="sd">        bgc_id: BGC identifier, e.g. MIBiG accession, GenBank accession.</span>
<a id="__codelineno-0-32" name="__codelineno-0-32"></a><span class="sd">        product_prediction: BGC&#39;s (predicted) natural products</span>
<a id="__codelineno-0-33" name="__codelineno-0-33"></a><span class="sd">            or product classes.</span>
<a id="__codelineno-0-34" name="__codelineno-0-34"></a>
<a id="__codelineno-0-35" name="__codelineno-0-35"></a><span class="sd">    Attributes:</span>
<a id="__codelineno-0-36" name="__codelineno-0-36"></a><span class="sd">        bgc_id: BGC identifier, e.g. MIBiG accession, GenBank accession.</span>
<a id="__codelineno-0-37" name="__codelineno-0-37"></a><span class="sd">        product_prediction: A tuple of (predicted) natural</span>
<a id="__codelineno-0-38" name="__codelineno-0-38"></a><span class="sd">            products or product classes of the BGC.</span>
<a id="__codelineno-0-39" name="__codelineno-0-39"></a><span class="sd">            For antiSMASH&#39;s GenBank data, the feature `region /product`</span>
<a id="__codelineno-0-40" name="__codelineno-0-40"></a><span class="sd">            gives product information.</span>
<a id="__codelineno-0-41" name="__codelineno-0-41"></a><span class="sd">            For MIBiG metadata, its biosynthetic class provides such info.</span>
<a id="__codelineno-0-42" name="__codelineno-0-42"></a><span class="sd">        mibig_bgc_class: A tuple of MIBiG biosynthetic</span>
<a id="__codelineno-0-43" name="__codelineno-0-43"></a><span class="sd">            classes to which the BGC belongs.</span>
<a id="__codelineno-0-44" name="__codelineno-0-44"></a><span class="sd">            Defaults to None.</span>
<a id="__codelineno-0-45" name="__codelineno-0-45"></a><span class="sd">            MIBiG defines 6 major biosynthetic classes for natural products,</span>
<a id="__codelineno-0-46" name="__codelineno-0-46"></a><span class="sd">            including &quot;NRP&quot;, &quot;Polyketide&quot;, &quot;RiPP&quot;, &quot;Terpene&quot;, &quot;Saccharide&quot;</span>
<a id="__codelineno-0-47" name="__codelineno-0-47"></a><span class="sd">            and &quot;Alkaloid&quot;. Note that natural products created by all other</span>
<a id="__codelineno-0-48" name="__codelineno-0-48"></a><span class="sd">            biosynthetic mechanisms fall under the category &quot;Other&quot;.</span>
<a id="__codelineno-0-49" name="__codelineno-0-49"></a><span class="sd">            More details see the publication: https://doi.org/10.1186/s40793-018-0318-y.</span>
<a id="__codelineno-0-50" name="__codelineno-0-50"></a><span class="sd">        description: Brief description of the BGC.</span>
<a id="__codelineno-0-51" name="__codelineno-0-51"></a><span class="sd">            Defaults to None.</span>
<a id="__codelineno-0-52" name="__codelineno-0-52"></a><span class="sd">        smiles: A tuple of SMILES formulas of the BGC&#39;s</span>
<a id="__codelineno-0-53" name="__codelineno-0-53"></a><span class="sd">            products.</span>
<a id="__codelineno-0-54" name="__codelineno-0-54"></a><span class="sd">            Defaults to None.</span>
<a id="__codelineno-0-55" name="__codelineno-0-55"></a><span class="sd">        antismash_file: The path to the antiSMASH GenBank file.</span>
<a id="__codelineno-0-56" name="__codelineno-0-56"></a><span class="sd">            Defaults to None.</span>
<a id="__codelineno-0-57" name="__codelineno-0-57"></a><span class="sd">        antismash_id: Identifier of the antiSMASH BGC, referring</span>
<a id="__codelineno-0-58" name="__codelineno-0-58"></a><span class="sd">            to the feature `VERSION` of GenBank file.</span>
<a id="__codelineno-0-59" name="__codelineno-0-59"></a><span class="sd">            Defaults to None.</span>
<a id="__codelineno-0-60" name="__codelineno-0-60"></a><span class="sd">        antismash_region: AntiSMASH BGC region number, referring</span>
<a id="__codelineno-0-61" name="__codelineno-0-61"></a><span class="sd">            to the feature `region` of GenBank file.</span>
<a id="__codelineno-0-62" name="__codelineno-0-62"></a><span class="sd">            Defaults to None.</span>
<a id="__codelineno-0-63" name="__codelineno-0-63"></a><span class="sd">        parents: The set of GCFs that contain the BGC.</span>
<a id="__codelineno-0-64" name="__codelineno-0-64"></a><span class="sd">        strain: The strain of the BGC.</span>
<a id="__codelineno-0-65" name="__codelineno-0-65"></a>
<a id="__codelineno-0-66" name="__codelineno-0-66"></a><span class="sd">    .. GenBank features:</span>
<a id="__codelineno-0-67" name="__codelineno-0-67"></a><span class="sd">        https://www.insdc.org/submitting-standards/feature-table/</span>
<a id="__codelineno-0-68" name="__codelineno-0-68"></a><span class="sd">    &quot;&quot;&quot;</span>
<a id="__codelineno-0-69" name="__codelineno-0-69"></a>    <span class="c1"># BGC metadata</span>
<a id="__codelineno-0-70" name="__codelineno-0-70"></a>    <span class="bp">self</span><span class="o">.</span><span class="n">bgc_id</span> <span class="o">=</span> <span class="n">bgc_id</span>
<a id="__codelineno-0-71" name="__codelineno-0-71"></a>    <span class="bp">self</span><span class="o">.</span><span class="n">product_prediction</span> <span class="o">=</span> <span class="n">product_prediction</span>
<a id="__codelineno-0-72" name="__codelineno-0-72"></a>
<a id="__codelineno-0-73" name="__codelineno-0-73"></a>    <span class="bp">self</span><span class="o">.</span><span class="n">mibig_bgc_class</span><span class="p">:</span> <span class="nb">tuple</span><span class="p">[</span><span class="nb">str</span><span class="p">]</span> <span class="o">|</span> <span class="kc">None</span> <span class="o">=</span> <span class="kc">None</span>
<a id="__codelineno-0-74" name="__codelineno-0-74"></a>    <span class="bp">self</span><span class="o">.</span><span class="n">description</span><span class="p">:</span> <span class="nb">str</span> <span class="o">|</span> <span class="kc">None</span> <span class="o">=</span> <span class="kc">None</span>
<a id="__codelineno-0-75" name="__codelineno-0-75"></a>    <span class="bp">self</span><span class="o">.</span><span class="n">smiles</span><span class="p">:</span> <span class="nb">tuple</span><span class="p">[</span><span class="nb">str</span><span class="p">]</span> <span class="o">|</span> <span class="kc">None</span> <span class="o">=</span> <span class="kc">None</span>
<a id="__codelineno-0-76" name="__codelineno-0-76"></a>
<a id="__codelineno-0-77" name="__codelineno-0-77"></a>    <span class="c1"># antismash related attributes</span>
<a id="__codelineno-0-78" name="__codelineno-0-78"></a>    <span class="bp">self</span><span class="o">.</span><span class="n">antismash_file</span><span class="p">:</span> <span class="nb">str</span> <span class="o">|</span> <span class="kc">None</span> <span class="o">=</span> <span class="kc">None</span>
<a id="__codelineno-0-79" name="__codelineno-0-79"></a>    <span class="bp">self</span><span class="o">.</span><span class="n">antismash_id</span><span class="p">:</span> <span class="nb">str</span> <span class="o">|</span> <span class="kc">None</span> <span class="o">=</span> <span class="kc">None</span>  <span class="c1"># version in .gbk, id in SeqRecord</span>
<a id="__codelineno-0-80" name="__codelineno-0-80"></a>    <span class="bp">self</span><span class="o">.</span><span class="n">antismash_region</span><span class="p">:</span> <span class="nb">int</span> <span class="o">|</span> <span class="kc">None</span> <span class="o">=</span> <span class="kc">None</span>  <span class="c1"># antismash region number</span>
<a id="__codelineno-0-81" name="__codelineno-0-81"></a>
<a id="__codelineno-0-82" name="__codelineno-0-82"></a>    <span class="c1"># other attributes</span>
<a id="__codelineno-0-83" name="__codelineno-0-83"></a>    <span class="bp">self</span><span class="o">.</span><span class="n">parents</span><span class="p">:</span> <span class="nb">set</span><span class="p">[</span><span class="n">GCF</span><span class="p">]</span> <span class="o">=</span> <span class="nb">set</span><span class="p">()</span>
<a id="__codelineno-0-84" name="__codelineno-0-84"></a>    <span class="bp">self</span><span class="o">.</span><span class="n">_strain</span><span class="p">:</span> <span class="n">Strain</span> <span class="o">|</span> <span class="kc">None</span> <span class="o">=</span> <span class="kc">None</span>
</code></pre></div></td></tr></table></div>
                </details>

  

  <div class="doc doc-children">







<div class="doc doc-object doc-attribute">



<h4 id="nplinker.genomics.BGC.bigscape_classes" class="doc doc-heading">
<code class="doc-symbol doc-symbol-heading doc-symbol-attribute"></code>          <span class="doc doc-object-name doc-attribute-name">bigscape_classes</span>

  
  <span class="doc doc-labels">
      <small class="doc doc-label doc-label-property"><code>property</code></small>
  </span>

<a href="#nplinker.genomics.BGC.bigscape_classes" class="headerlink" title="Permanent link">&para;</a></h4>
<div class="language-python doc-signature highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="n">bigscape_classes</span><span class="p">:</span> <span class="n"><a class="autorefs autorefs-external" href="https://docs.python.org/3/library/stdtypes.html#set">set</a></span><span class="p">[</span><span class="n"><a class="autorefs autorefs-external" href="https://docs.python.org/3/library/stdtypes.html#str">str</a></span> <span class="o">|</span> <span class="kc">None</span><span class="p">]</span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Get BiG-SCAPE's BGC classes.</p>
<p>BiG-SCAPE's BGC classes are similar to those defined in MiBIG but have
more categories (7 classes). More details see:
<a href="https://doi.org/10.1038%2Fs41589-019-0400-9">https://doi.org/10.1038%2Fs41589-019-0400-9</a>.</p>
  </div>

</div>

<div class="doc doc-object doc-attribute">



<h4 id="nplinker.genomics.BGC.aa_predictions" class="doc doc-heading">
<code class="doc-symbol doc-symbol-heading doc-symbol-attribute"></code>          <span class="doc doc-object-name doc-attribute-name">aa_predictions</span>

  
  <span class="doc doc-labels">
      <small class="doc doc-label doc-label-property"><code>property</code></small>
  </span>

<a href="#nplinker.genomics.BGC.aa_predictions" class="headerlink" title="Permanent link">&para;</a></h4>
<div class="language-python doc-signature highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="n">aa_predictions</span><span class="p">:</span> <span class="n"><a class="autorefs autorefs-external" href="https://docs.python.org/3/library/stdtypes.html#list">list</a></span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Amino acids as predicted monomers of product.</p>



  <p><strong>Returns:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Type</th>
        <th>Description</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td>
                <code><a class="autorefs autorefs-external" href="https://docs.python.org/3/library/stdtypes.html#list">list</a></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>list of dicts with key as amino acid and value as prediction</p>
            </div>
          </td>
        </tr>
        <tr>
          <td>
                <code><a class="autorefs autorefs-external" href="https://docs.python.org/3/library/stdtypes.html#list">list</a></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>probability.</p>
            </div>
          </td>
        </tr>
    </tbody>
  </table>
  </div>

</div>




<div class="doc doc-object doc-function">



<h4 id="nplinker.genomics.BGC.add_parent" class="doc doc-heading">
<code class="doc-symbol doc-symbol-heading doc-symbol-method"></code>          <span class="doc doc-object-name doc-function-name">add_parent</span>


<a href="#nplinker.genomics.BGC.add_parent" class="headerlink" title="Permanent link">&para;</a></h4>
<div class="language-python doc-signature highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="n">add_parent</span><span class="p">(</span><span class="n">gcf</span><span class="p">:</span> <span class="n"><a class="autorefs autorefs-internal" title="nplinker.genomics.gcf.GCF" href="#nplinker.genomics.GCF">GCF</a></span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Add a parent GCF to the BGC.</p>



  <p><strong>Parameters:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Name</th>
        <th>Type</th>
        <th>Description</th>
        <th>Default</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td><code>gcf</code></td>
          <td>
                <code><a class="autorefs autorefs-internal" title="nplinker.genomics.gcf.GCF" href="#nplinker.genomics.GCF">GCF</a></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>gene cluster family</p>
            </div>
          </td>
          <td>
              <em>required</em>
          </td>
        </tr>
    </tbody>
  </table>

          <details class="quote">
            <summary>Source code in <code>src/nplinker/genomics/bgc.py</code></summary>
            <div class="language-python highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-0-108">108</a></span>
<span class="normal"><a href="#__codelineno-0-109">109</a></span>
<span class="normal"><a href="#__codelineno-0-110">110</a></span>
<span class="normal"><a href="#__codelineno-0-111">111</a></span>
<span class="normal"><a href="#__codelineno-0-112">112</a></span>
<span class="normal"><a href="#__codelineno-0-113">113</a></span>
<span class="normal"><a href="#__codelineno-0-114">114</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-0-108" name="__codelineno-0-108"></a><span class="k">def</span> <span class="nf">add_parent</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">gcf</span><span class="p">:</span> <span class="n">GCF</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
<a id="__codelineno-0-109" name="__codelineno-0-109"></a><span class="w">    </span><span class="sd">&quot;&quot;&quot;Add a parent GCF to the BGC.</span>
<a id="__codelineno-0-110" name="__codelineno-0-110"></a>
<a id="__codelineno-0-111" name="__codelineno-0-111"></a><span class="sd">    Args:</span>
<a id="__codelineno-0-112" name="__codelineno-0-112"></a><span class="sd">        gcf: gene cluster family</span>
<a id="__codelineno-0-113" name="__codelineno-0-113"></a><span class="sd">    &quot;&quot;&quot;</span>
<a id="__codelineno-0-114" name="__codelineno-0-114"></a>    <span class="n">gcf</span><span class="o">.</span><span class="n">add_bgc</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span>
</code></pre></div></td></tr></table></div>
          </details>
  </div>

</div>


<div class="doc doc-object doc-function">



<h4 id="nplinker.genomics.BGC.detach_parent" class="doc doc-heading">
<code class="doc-symbol doc-symbol-heading doc-symbol-method"></code>          <span class="doc doc-object-name doc-function-name">detach_parent</span>


<a href="#nplinker.genomics.BGC.detach_parent" class="headerlink" title="Permanent link">&para;</a></h4>
<div class="language-python doc-signature highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="n">detach_parent</span><span class="p">(</span><span class="n">gcf</span><span class="p">:</span> <span class="n"><a class="autorefs autorefs-internal" title="nplinker.genomics.gcf.GCF" href="#nplinker.genomics.GCF">GCF</a></span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Remove a parent GCF.</p>

          <details class="quote">
            <summary>Source code in <code>src/nplinker/genomics/bgc.py</code></summary>
            <div class="language-python highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-0-116">116</a></span>
<span class="normal"><a href="#__codelineno-0-117">117</a></span>
<span class="normal"><a href="#__codelineno-0-118">118</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-0-116" name="__codelineno-0-116"></a><span class="k">def</span> <span class="nf">detach_parent</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">gcf</span><span class="p">:</span> <span class="n">GCF</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
<a id="__codelineno-0-117" name="__codelineno-0-117"></a><span class="w">    </span><span class="sd">&quot;&quot;&quot;Remove a parent GCF.&quot;&quot;&quot;</span>
<a id="__codelineno-0-118" name="__codelineno-0-118"></a>    <span class="n">gcf</span><span class="o">.</span><span class="n">detach_bgc</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span>
</code></pre></div></td></tr></table></div>
          </details>
  </div>

</div>


<div class="doc doc-object doc-function">



<h4 id="nplinker.genomics.BGC.is_mibig" class="doc doc-heading">
<code class="doc-symbol doc-symbol-heading doc-symbol-method"></code>          <span class="doc doc-object-name doc-function-name">is_mibig</span>


<a href="#nplinker.genomics.BGC.is_mibig" class="headerlink" title="Permanent link">&para;</a></h4>
<div class="language-python doc-signature highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="n">is_mibig</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n"><a class="autorefs autorefs-external" href="https://docs.python.org/3/library/functions.html#bool">bool</a></span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Check if the BGC is MIBiG reference BGC or not.</p>

<details class="note" open>
  <summary>Note</summary>
  <p>This method evaluates MIBiG BGC based on the pattern that MIBiG
BGC names start with "BGC". It might give false positive result.</p>
</details>


  <p><strong>Returns:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Type</th>
        <th>Description</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td>
                <code><a class="autorefs autorefs-external" href="https://docs.python.org/3/library/functions.html#bool">bool</a></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>True if it's MIBiG reference BGC</p>
            </div>
          </td>
        </tr>
    </tbody>
  </table>

          <details class="quote">
            <summary>Source code in <code>src/nplinker/genomics/bgc.py</code></summary>
            <div class="language-python highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-0-138">138</a></span>
<span class="normal"><a href="#__codelineno-0-139">139</a></span>
<span class="normal"><a href="#__codelineno-0-140">140</a></span>
<span class="normal"><a href="#__codelineno-0-141">141</a></span>
<span class="normal"><a href="#__codelineno-0-142">142</a></span>
<span class="normal"><a href="#__codelineno-0-143">143</a></span>
<span class="normal"><a href="#__codelineno-0-144">144</a></span>
<span class="normal"><a href="#__codelineno-0-145">145</a></span>
<span class="normal"><a href="#__codelineno-0-146">146</a></span>
<span class="normal"><a href="#__codelineno-0-147">147</a></span>
<span class="normal"><a href="#__codelineno-0-148">148</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-0-138" name="__codelineno-0-138"></a><span class="k">def</span> <span class="nf">is_mibig</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">bool</span><span class="p">:</span>
<a id="__codelineno-0-139" name="__codelineno-0-139"></a><span class="w">    </span><span class="sd">&quot;&quot;&quot;Check if the BGC is MIBiG reference BGC or not.</span>
<a id="__codelineno-0-140" name="__codelineno-0-140"></a>
<a id="__codelineno-0-141" name="__codelineno-0-141"></a><span class="sd">    Note:</span>
<a id="__codelineno-0-142" name="__codelineno-0-142"></a><span class="sd">        This method evaluates MIBiG BGC based on the pattern that MIBiG</span>
<a id="__codelineno-0-143" name="__codelineno-0-143"></a><span class="sd">        BGC names start with &quot;BGC&quot;. It might give false positive result.</span>
<a id="__codelineno-0-144" name="__codelineno-0-144"></a>
<a id="__codelineno-0-145" name="__codelineno-0-145"></a><span class="sd">    Returns:</span>
<a id="__codelineno-0-146" name="__codelineno-0-146"></a><span class="sd">        True if it&#39;s MIBiG reference BGC</span>
<a id="__codelineno-0-147" name="__codelineno-0-147"></a><span class="sd">    &quot;&quot;&quot;</span>
<a id="__codelineno-0-148" name="__codelineno-0-148"></a>    <span class="k">return</span> <span class="bp">self</span><span class="o">.</span><span class="n">bgc_id</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s2">&quot;BGC&quot;</span><span class="p">)</span>
</code></pre></div></td></tr></table></div>
          </details>
  </div>

</div>



  </div>

  </div>


</div>

<div class="doc doc-object doc-class">



<h3 id="nplinker.genomics.GCF" class="doc doc-heading">
<code class="doc-symbol doc-symbol-heading doc-symbol-class"></code>          <span class="doc doc-object-name doc-class-name">GCF</span>


<a href="#nplinker.genomics.GCF" class="headerlink" title="Permanent link">&para;</a></h3>
<div class="language-python doc-signature highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="n">GCF</span><span class="p">(</span><span class="n">gcf_id</span><span class="p">:</span> <span class="n"><a class="autorefs autorefs-external" href="https://docs.python.org/3/library/stdtypes.html#str">str</a></span><span class="p">)</span>
</code></pre></div>

  <div class="doc doc-contents ">

  
      <p>Class to model gene cluster family (GCF).</p>
<p>GCF is a group of similar BGCs and generated by clustering BGCs with
tools such as BiG-SCAPE and BiG-SLICE.</p>



  <p><strong>Parameters:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Name</th>
        <th>Type</th>
        <th>Description</th>
        <th>Default</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td><code>gcf_id</code></td>
          <td>
                <code><a class="autorefs autorefs-external" href="https://docs.python.org/3/library/stdtypes.html#str">str</a></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>id of the GCF object.</p>
            </div>
          </td>
          <td>
              <em>required</em>
          </td>
        </tr>
    </tbody>
  </table>



  <p><strong>Attributes:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Name</th>
        <th>Type</th>
        <th>Description</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td><code><span title="nplinker.genomics.GCF.gcf_id">gcf_id</span></code></td>
          <td>
          </td>
          <td>
            <div class="doc-md-description">
              <p>id of the GCF object.</p>
            </div>
          </td>
        </tr>
        <tr>
          <td><code><span title="nplinker.genomics.GCF.bgc_ids">bgc_ids</span></code></td>
          <td>
          </td>
          <td>
            <div class="doc-md-description">
              <p>a set of BGC ids that belongs to the GCF.</p>
            </div>
          </td>
        </tr>
        <tr>
          <td><code><span title="nplinker.genomics.GCF.bigscape_class">bigscape_class</span></code></td>
          <td>
          </td>
          <td>
            <div class="doc-md-description">
              <p>BiG-SCAPE's BGC class.
BiG-SCAPE's BGC classes are similar to those defined in MiBIG
but have more categories (7 classes). More details see:
<a href="https://doi.org/10.1038%2Fs41589-019-0400-9">https://doi.org/10.1038%2Fs41589-019-0400-9</a>.</p>
            </div>
          </td>
        </tr>
    </tbody>
  </table>

                <details class="quote">
                  <summary>Source code in <code>src/nplinker/genomics/gcf.py</code></summary>
                  <div class="language-python highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-0-15">15</a></span>
<span class="normal"><a href="#__codelineno-0-16">16</a></span>
<span class="normal"><a href="#__codelineno-0-17">17</a></span>
<span class="normal"><a href="#__codelineno-0-18">18</a></span>
<span class="normal"><a href="#__codelineno-0-19">19</a></span>
<span class="normal"><a href="#__codelineno-0-20">20</a></span>
<span class="normal"><a href="#__codelineno-0-21">21</a></span>
<span class="normal"><a href="#__codelineno-0-22">22</a></span>
<span class="normal"><a href="#__codelineno-0-23">23</a></span>
<span class="normal"><a href="#__codelineno-0-24">24</a></span>
<span class="normal"><a href="#__codelineno-0-25">25</a></span>
<span class="normal"><a href="#__codelineno-0-26">26</a></span>
<span class="normal"><a href="#__codelineno-0-27">27</a></span>
<span class="normal"><a href="#__codelineno-0-28">28</a></span>
<span class="normal"><a href="#__codelineno-0-29">29</a></span>
<span class="normal"><a href="#__codelineno-0-30">30</a></span>
<span class="normal"><a href="#__codelineno-0-31">31</a></span>
<span class="normal"><a href="#__codelineno-0-32">32</a></span>
<span class="normal"><a href="#__codelineno-0-33">33</a></span>
<span class="normal"><a href="#__codelineno-0-34">34</a></span>
<span class="normal"><a href="#__codelineno-0-35">35</a></span>
<span class="normal"><a href="#__codelineno-0-36">36</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-0-15" name="__codelineno-0-15"></a><span class="k">def</span> <span class="fm">__init__</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">gcf_id</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="o">/</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
<a id="__codelineno-0-16" name="__codelineno-0-16"></a><span class="w">    </span><span class="sd">&quot;&quot;&quot;Class to model gene cluster family (GCF).</span>
<a id="__codelineno-0-17" name="__codelineno-0-17"></a>
<a id="__codelineno-0-18" name="__codelineno-0-18"></a><span class="sd">    GCF is a group of similar BGCs and generated by clustering BGCs with</span>
<a id="__codelineno-0-19" name="__codelineno-0-19"></a><span class="sd">    tools such as BiG-SCAPE and BiG-SLICE.</span>
<a id="__codelineno-0-20" name="__codelineno-0-20"></a>
<a id="__codelineno-0-21" name="__codelineno-0-21"></a><span class="sd">    Args:</span>
<a id="__codelineno-0-22" name="__codelineno-0-22"></a><span class="sd">        gcf_id: id of the GCF object.</span>
<a id="__codelineno-0-23" name="__codelineno-0-23"></a>
<a id="__codelineno-0-24" name="__codelineno-0-24"></a><span class="sd">    Attributes:</span>
<a id="__codelineno-0-25" name="__codelineno-0-25"></a><span class="sd">        gcf_id: id of the GCF object.</span>
<a id="__codelineno-0-26" name="__codelineno-0-26"></a><span class="sd">        bgc_ids: a set of BGC ids that belongs to the GCF.</span>
<a id="__codelineno-0-27" name="__codelineno-0-27"></a><span class="sd">        bigscape_class: BiG-SCAPE&#39;s BGC class.</span>
<a id="__codelineno-0-28" name="__codelineno-0-28"></a><span class="sd">            BiG-SCAPE&#39;s BGC classes are similar to those defined in MiBIG</span>
<a id="__codelineno-0-29" name="__codelineno-0-29"></a><span class="sd">            but have more categories (7 classes). More details see:</span>
<a id="__codelineno-0-30" name="__codelineno-0-30"></a><span class="sd">            https://doi.org/10.1038%2Fs41589-019-0400-9.</span>
<a id="__codelineno-0-31" name="__codelineno-0-31"></a><span class="sd">    &quot;&quot;&quot;</span>
<a id="__codelineno-0-32" name="__codelineno-0-32"></a>    <span class="bp">self</span><span class="o">.</span><span class="n">gcf_id</span> <span class="o">=</span> <span class="n">gcf_id</span>
<a id="__codelineno-0-33" name="__codelineno-0-33"></a>    <span class="bp">self</span><span class="o">.</span><span class="n">bgc_ids</span><span class="p">:</span> <span class="nb">set</span><span class="p">[</span><span class="nb">str</span><span class="p">]</span> <span class="o">=</span> <span class="nb">set</span><span class="p">()</span>
<a id="__codelineno-0-34" name="__codelineno-0-34"></a>    <span class="bp">self</span><span class="o">.</span><span class="n">bigscape_class</span><span class="p">:</span> <span class="nb">str</span> <span class="o">|</span> <span class="kc">None</span> <span class="o">=</span> <span class="kc">None</span>
<a id="__codelineno-0-35" name="__codelineno-0-35"></a>    <span class="bp">self</span><span class="o">.</span><span class="n">_bgcs</span><span class="p">:</span> <span class="nb">set</span><span class="p">[</span><span class="n">BGC</span><span class="p">]</span> <span class="o">=</span> <span class="nb">set</span><span class="p">()</span>
<a id="__codelineno-0-36" name="__codelineno-0-36"></a>    <span class="bp">self</span><span class="o">.</span><span class="n">_strains</span><span class="p">:</span> <span class="n">StrainCollection</span> <span class="o">=</span> <span class="n">StrainCollection</span><span class="p">()</span>
</code></pre></div></td></tr></table></div>
                </details>

  

  <div class="doc doc-children">







<div class="doc doc-object doc-attribute">



<h4 id="nplinker.genomics.GCF.bgcs" class="doc doc-heading">
<code class="doc-symbol doc-symbol-heading doc-symbol-attribute"></code>          <span class="doc doc-object-name doc-attribute-name">bgcs</span>

  
  <span class="doc doc-labels">
      <small class="doc doc-label doc-label-property"><code>property</code></small>
  </span>

<a href="#nplinker.genomics.GCF.bgcs" class="headerlink" title="Permanent link">&para;</a></h4>
<div class="language-python doc-signature highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="n">bgcs</span><span class="p">:</span> <span class="n"><a class="autorefs autorefs-external" href="https://docs.python.org/3/library/stdtypes.html#set">set</a></span><span class="p">[</span><span class="n"><a class="autorefs autorefs-internal" title="nplinker.genomics.bgc.BGC" href="#nplinker.genomics.BGC">BGC</a></span><span class="p">]</span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Get the BGC objects.</p>
  </div>

</div>

<div class="doc doc-object doc-attribute">



<h4 id="nplinker.genomics.GCF.strains" class="doc doc-heading">
<code class="doc-symbol doc-symbol-heading doc-symbol-attribute"></code>          <span class="doc doc-object-name doc-attribute-name">strains</span>

  
  <span class="doc doc-labels">
      <small class="doc doc-label doc-label-property"><code>property</code></small>
  </span>

<a href="#nplinker.genomics.GCF.strains" class="headerlink" title="Permanent link">&para;</a></h4>
<div class="language-python doc-signature highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="n">strains</span><span class="p">:</span> <span class="n"><a class="autorefs autorefs-internal" title="nplinker.strain.StrainCollection" href="../strain/#nplinker.strain.StrainCollection">StrainCollection</a></span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Get the strains in the GCF.</p>
  </div>

</div>




<div class="doc doc-object doc-function">



<h4 id="nplinker.genomics.GCF.add_bgc" class="doc doc-heading">
<code class="doc-symbol doc-symbol-heading doc-symbol-method"></code>          <span class="doc doc-object-name doc-function-name">add_bgc</span>


<a href="#nplinker.genomics.GCF.add_bgc" class="headerlink" title="Permanent link">&para;</a></h4>
<div class="language-python doc-signature highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="n">add_bgc</span><span class="p">(</span><span class="n">bgc</span><span class="p">:</span> <span class="n"><a class="autorefs autorefs-internal" title="nplinker.genomics.bgc.BGC" href="#nplinker.genomics.BGC">BGC</a></span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Add a BGC object to the GCF.</p>

          <details class="quote">
            <summary>Source code in <code>src/nplinker/genomics/gcf.py</code></summary>
            <div class="language-python highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-0-70">70</a></span>
<span class="normal"><a href="#__codelineno-0-71">71</a></span>
<span class="normal"><a href="#__codelineno-0-72">72</a></span>
<span class="normal"><a href="#__codelineno-0-73">73</a></span>
<span class="normal"><a href="#__codelineno-0-74">74</a></span>
<span class="normal"><a href="#__codelineno-0-75">75</a></span>
<span class="normal"><a href="#__codelineno-0-76">76</a></span>
<span class="normal"><a href="#__codelineno-0-77">77</a></span>
<span class="normal"><a href="#__codelineno-0-78">78</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-0-70" name="__codelineno-0-70"></a><span class="k">def</span> <span class="nf">add_bgc</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">bgc</span><span class="p">:</span> <span class="n">BGC</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
<a id="__codelineno-0-71" name="__codelineno-0-71"></a><span class="w">    </span><span class="sd">&quot;&quot;&quot;Add a BGC object to the GCF.&quot;&quot;&quot;</span>
<a id="__codelineno-0-72" name="__codelineno-0-72"></a>    <span class="n">bgc</span><span class="o">.</span><span class="n">parents</span><span class="o">.</span><span class="n">add</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span>
<a id="__codelineno-0-73" name="__codelineno-0-73"></a>    <span class="bp">self</span><span class="o">.</span><span class="n">_bgcs</span><span class="o">.</span><span class="n">add</span><span class="p">(</span><span class="n">bgc</span><span class="p">)</span>
<a id="__codelineno-0-74" name="__codelineno-0-74"></a>    <span class="bp">self</span><span class="o">.</span><span class="n">bgc_ids</span><span class="o">.</span><span class="n">add</span><span class="p">(</span><span class="n">bgc</span><span class="o">.</span><span class="n">bgc_id</span><span class="p">)</span>
<a id="__codelineno-0-75" name="__codelineno-0-75"></a>    <span class="k">if</span> <span class="n">bgc</span><span class="o">.</span><span class="n">strain</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
<a id="__codelineno-0-76" name="__codelineno-0-76"></a>        <span class="bp">self</span><span class="o">.</span><span class="n">_strains</span><span class="o">.</span><span class="n">add</span><span class="p">(</span><span class="n">bgc</span><span class="o">.</span><span class="n">strain</span><span class="p">)</span>
<a id="__codelineno-0-77" name="__codelineno-0-77"></a>    <span class="k">else</span><span class="p">:</span>
<a id="__codelineno-0-78" name="__codelineno-0-78"></a>        <span class="n">logger</span><span class="o">.</span><span class="n">warning</span><span class="p">(</span><span class="s2">&quot;No strain specified for the BGC </span><span class="si">%s</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">bgc</span><span class="o">.</span><span class="n">bgc_id</span><span class="p">)</span>
</code></pre></div></td></tr></table></div>
          </details>
  </div>

</div>


<div class="doc doc-object doc-function">



<h4 id="nplinker.genomics.GCF.detach_bgc" class="doc doc-heading">
<code class="doc-symbol doc-symbol-heading doc-symbol-method"></code>          <span class="doc doc-object-name doc-function-name">detach_bgc</span>


<a href="#nplinker.genomics.GCF.detach_bgc" class="headerlink" title="Permanent link">&para;</a></h4>
<div class="language-python doc-signature highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="n">detach_bgc</span><span class="p">(</span><span class="n">bgc</span><span class="p">:</span> <span class="n"><a class="autorefs autorefs-internal" title="nplinker.genomics.bgc.BGC" href="#nplinker.genomics.BGC">BGC</a></span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Remove a child BGC object.</p>

          <details class="quote">
            <summary>Source code in <code>src/nplinker/genomics/gcf.py</code></summary>
            <div class="language-python highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-0-80">80</a></span>
<span class="normal"><a href="#__codelineno-0-81">81</a></span>
<span class="normal"><a href="#__codelineno-0-82">82</a></span>
<span class="normal"><a href="#__codelineno-0-83">83</a></span>
<span class="normal"><a href="#__codelineno-0-84">84</a></span>
<span class="normal"><a href="#__codelineno-0-85">85</a></span>
<span class="normal"><a href="#__codelineno-0-86">86</a></span>
<span class="normal"><a href="#__codelineno-0-87">87</a></span>
<span class="normal"><a href="#__codelineno-0-88">88</a></span>
<span class="normal"><a href="#__codelineno-0-89">89</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-0-80" name="__codelineno-0-80"></a><span class="k">def</span> <span class="nf">detach_bgc</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">bgc</span><span class="p">:</span> <span class="n">BGC</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="kc">None</span><span class="p">:</span>
<a id="__codelineno-0-81" name="__codelineno-0-81"></a><span class="w">    </span><span class="sd">&quot;&quot;&quot;Remove a child BGC object.&quot;&quot;&quot;</span>
<a id="__codelineno-0-82" name="__codelineno-0-82"></a>    <span class="n">bgc</span><span class="o">.</span><span class="n">parents</span><span class="o">.</span><span class="n">remove</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span>
<a id="__codelineno-0-83" name="__codelineno-0-83"></a>    <span class="bp">self</span><span class="o">.</span><span class="n">_bgcs</span><span class="o">.</span><span class="n">remove</span><span class="p">(</span><span class="n">bgc</span><span class="p">)</span>
<a id="__codelineno-0-84" name="__codelineno-0-84"></a>    <span class="bp">self</span><span class="o">.</span><span class="n">bgc_ids</span><span class="o">.</span><span class="n">remove</span><span class="p">(</span><span class="n">bgc</span><span class="o">.</span><span class="n">bgc_id</span><span class="p">)</span>
<a id="__codelineno-0-85" name="__codelineno-0-85"></a>    <span class="k">if</span> <span class="n">bgc</span><span class="o">.</span><span class="n">strain</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
<a id="__codelineno-0-86" name="__codelineno-0-86"></a>        <span class="k">for</span> <span class="n">other_bgc</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">_bgcs</span><span class="p">:</span>
<a id="__codelineno-0-87" name="__codelineno-0-87"></a>            <span class="k">if</span> <span class="n">other_bgc</span><span class="o">.</span><span class="n">strain</span> <span class="o">==</span> <span class="n">bgc</span><span class="o">.</span><span class="n">strain</span><span class="p">:</span>
<a id="__codelineno-0-88" name="__codelineno-0-88"></a>                <span class="k">return</span>
<a id="__codelineno-0-89" name="__codelineno-0-89"></a>        <span class="bp">self</span><span class="o">.</span><span class="n">_strains</span><span class="o">.</span><span class="n">remove</span><span class="p">(</span><span class="n">bgc</span><span class="o">.</span><span class="n">strain</span><span class="p">)</span>
</code></pre></div></td></tr></table></div>
          </details>
  </div>

</div>


<div class="doc doc-object doc-function">



<h4 id="nplinker.genomics.GCF.has_strain" class="doc doc-heading">
<code class="doc-symbol doc-symbol-heading doc-symbol-method"></code>          <span class="doc doc-object-name doc-function-name">has_strain</span>


<a href="#nplinker.genomics.GCF.has_strain" class="headerlink" title="Permanent link">&para;</a></h4>
<div class="language-python doc-signature highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="n">has_strain</span><span class="p">(</span><span class="n">strain</span><span class="p">:</span> <span class="n"><a class="autorefs autorefs-internal" title="nplinker.strain.Strain" href="../strain/#nplinker.strain.Strain">Strain</a></span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n"><a class="autorefs autorefs-external" href="https://docs.python.org/3/library/functions.html#bool">bool</a></span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Check if the given strain exists.</p>



  <p><strong>Parameters:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Name</th>
        <th>Type</th>
        <th>Description</th>
        <th>Default</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td><code>strain</code></td>
          <td>
                <code><a class="autorefs autorefs-internal" title="nplinker.strain.Strain" href="../strain/#nplinker.strain.Strain">Strain</a></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p><code>Strain</code> object.</p>
            </div>
          </td>
          <td>
              <em>required</em>
          </td>
        </tr>
    </tbody>
  </table>



  <p><strong>Returns:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Type</th>
        <th>Description</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td>
                <code><a class="autorefs autorefs-external" href="https://docs.python.org/3/library/functions.html#bool">bool</a></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>True when the given strain exist.</p>
            </div>
          </td>
        </tr>
    </tbody>
  </table>

          <details class="quote">
            <summary>Source code in <code>src/nplinker/genomics/gcf.py</code></summary>
            <div class="language-python highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-0-91"> 91</a></span>
<span class="normal"><a href="#__codelineno-0-92"> 92</a></span>
<span class="normal"><a href="#__codelineno-0-93"> 93</a></span>
<span class="normal"><a href="#__codelineno-0-94"> 94</a></span>
<span class="normal"><a href="#__codelineno-0-95"> 95</a></span>
<span class="normal"><a href="#__codelineno-0-96"> 96</a></span>
<span class="normal"><a href="#__codelineno-0-97"> 97</a></span>
<span class="normal"><a href="#__codelineno-0-98"> 98</a></span>
<span class="normal"><a href="#__codelineno-0-99"> 99</a></span>
<span class="normal"><a href="#__codelineno-0-100">100</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-0-91" name="__codelineno-0-91"></a><span class="k">def</span> <span class="nf">has_strain</span><span class="p">(</span><span class="bp">self</span><span class="p">,</span> <span class="n">strain</span><span class="p">:</span> <span class="n">Strain</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">bool</span><span class="p">:</span>
<a id="__codelineno-0-92" name="__codelineno-0-92"></a><span class="w">    </span><span class="sd">&quot;&quot;&quot;Check if the given strain exists.</span>
<a id="__codelineno-0-93" name="__codelineno-0-93"></a>
<a id="__codelineno-0-94" name="__codelineno-0-94"></a><span class="sd">    Args:</span>
<a id="__codelineno-0-95" name="__codelineno-0-95"></a><span class="sd">        strain: `Strain` object.</span>
<a id="__codelineno-0-96" name="__codelineno-0-96"></a>
<a id="__codelineno-0-97" name="__codelineno-0-97"></a><span class="sd">    Returns:</span>
<a id="__codelineno-0-98" name="__codelineno-0-98"></a><span class="sd">        True when the given strain exist.</span>
<a id="__codelineno-0-99" name="__codelineno-0-99"></a><span class="sd">    &quot;&quot;&quot;</span>
<a id="__codelineno-0-100" name="__codelineno-0-100"></a>    <span class="k">return</span> <span class="n">strain</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">_strains</span>
</code></pre></div></td></tr></table></div>
          </details>
  </div>

</div>


<div class="doc doc-object doc-function">



<h4 id="nplinker.genomics.GCF.has_mibig_only" class="doc doc-heading">
<code class="doc-symbol doc-symbol-heading doc-symbol-method"></code>          <span class="doc doc-object-name doc-function-name">has_mibig_only</span>


<a href="#nplinker.genomics.GCF.has_mibig_only" class="headerlink" title="Permanent link">&para;</a></h4>
<div class="language-python doc-signature highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="n">has_mibig_only</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n"><a class="autorefs autorefs-external" href="https://docs.python.org/3/library/functions.html#bool">bool</a></span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Check if the GCF's children are only MIBiG BGCs.</p>



  <p><strong>Returns:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Type</th>
        <th>Description</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td>
                <code><a class="autorefs autorefs-external" href="https://docs.python.org/3/library/functions.html#bool">bool</a></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>True if <code>GCF.bgc_ids</code> are only MIBiG BGC ids.</p>
            </div>
          </td>
        </tr>
    </tbody>
  </table>

          <details class="quote">
            <summary>Source code in <code>src/nplinker/genomics/gcf.py</code></summary>
            <div class="language-python highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-0-102">102</a></span>
<span class="normal"><a href="#__codelineno-0-103">103</a></span>
<span class="normal"><a href="#__codelineno-0-104">104</a></span>
<span class="normal"><a href="#__codelineno-0-105">105</a></span>
<span class="normal"><a href="#__codelineno-0-106">106</a></span>
<span class="normal"><a href="#__codelineno-0-107">107</a></span>
<span class="normal"><a href="#__codelineno-0-108">108</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-0-102" name="__codelineno-0-102"></a><span class="k">def</span> <span class="nf">has_mibig_only</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">bool</span><span class="p">:</span>
<a id="__codelineno-0-103" name="__codelineno-0-103"></a><span class="w">    </span><span class="sd">&quot;&quot;&quot;Check if the GCF&#39;s children are only MIBiG BGCs.</span>
<a id="__codelineno-0-104" name="__codelineno-0-104"></a>
<a id="__codelineno-0-105" name="__codelineno-0-105"></a><span class="sd">    Returns:</span>
<a id="__codelineno-0-106" name="__codelineno-0-106"></a><span class="sd">        True if `GCF.bgc_ids` are only MIBiG BGC ids.</span>
<a id="__codelineno-0-107" name="__codelineno-0-107"></a><span class="sd">    &quot;&quot;&quot;</span>
<a id="__codelineno-0-108" name="__codelineno-0-108"></a>    <span class="k">return</span> <span class="nb">all</span><span class="p">(</span><span class="nb">map</span><span class="p">(</span><span class="k">lambda</span> <span class="nb">id</span><span class="p">:</span> <span class="nb">id</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s2">&quot;BGC&quot;</span><span class="p">),</span> <span class="bp">self</span><span class="o">.</span><span class="n">bgc_ids</span><span class="p">))</span>
</code></pre></div></td></tr></table></div>
          </details>
  </div>

</div>


<div class="doc doc-object doc-function">



<h4 id="nplinker.genomics.GCF.is_singleton" class="doc doc-heading">
<code class="doc-symbol doc-symbol-heading doc-symbol-method"></code>          <span class="doc doc-object-name doc-function-name">is_singleton</span>


<a href="#nplinker.genomics.GCF.is_singleton" class="headerlink" title="Permanent link">&para;</a></h4>
<div class="language-python doc-signature highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="n">is_singleton</span><span class="p">()</span> <span class="o">-&gt;</span> <span class="n"><a class="autorefs autorefs-external" href="https://docs.python.org/3/library/functions.html#bool">bool</a></span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Check if the GCF contains only one BGC.</p>



  <p><strong>Returns:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Type</th>
        <th>Description</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td>
                <code><a class="autorefs autorefs-external" href="https://docs.python.org/3/library/functions.html#bool">bool</a></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>True if <code>GCF.bgc_ids</code> contains only one BGC id.</p>
            </div>
          </td>
        </tr>
    </tbody>
  </table>

          <details class="quote">
            <summary>Source code in <code>src/nplinker/genomics/gcf.py</code></summary>
            <div class="language-python highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-0-110">110</a></span>
<span class="normal"><a href="#__codelineno-0-111">111</a></span>
<span class="normal"><a href="#__codelineno-0-112">112</a></span>
<span class="normal"><a href="#__codelineno-0-113">113</a></span>
<span class="normal"><a href="#__codelineno-0-114">114</a></span>
<span class="normal"><a href="#__codelineno-0-115">115</a></span>
<span class="normal"><a href="#__codelineno-0-116">116</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-0-110" name="__codelineno-0-110"></a><span class="k">def</span> <span class="nf">is_singleton</span><span class="p">(</span><span class="bp">self</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">bool</span><span class="p">:</span>
<a id="__codelineno-0-111" name="__codelineno-0-111"></a><span class="w">    </span><span class="sd">&quot;&quot;&quot;Check if the GCF contains only one BGC.</span>
<a id="__codelineno-0-112" name="__codelineno-0-112"></a>
<a id="__codelineno-0-113" name="__codelineno-0-113"></a><span class="sd">    Returns:</span>
<a id="__codelineno-0-114" name="__codelineno-0-114"></a><span class="sd">        True if `GCF.bgc_ids` contains only one BGC id.</span>
<a id="__codelineno-0-115" name="__codelineno-0-115"></a><span class="sd">    &quot;&quot;&quot;</span>
<a id="__codelineno-0-116" name="__codelineno-0-116"></a>    <span class="k">return</span> <span class="nb">len</span><span class="p">(</span><span class="bp">self</span><span class="o">.</span><span class="n">bgc_ids</span><span class="p">)</span> <span class="o">==</span> <span class="mi">1</span>
</code></pre></div></td></tr></table></div>
          </details>
  </div>

</div>



  </div>

  </div>


</div>




  </div>

  </div>

</div>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var tab,labels=set.querySelector(".tabbed-labels");for(tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["content.tabs.link", "content.code.annotate", "content.code.copy", "announce.dismiss", "navigation.tabs"], "search": "../../assets/javascripts/workers/search.b8dbb3d2.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"provider": "mike"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.c8d2eff1.min.js"></script>
      
        <script src="../../extra/fluff.js"></script>
      
        <script src="https://samuelcolvin.github.io/mkdocs-run-code/run_code_main.js"></script>
      
    
  </body>
</html>